### PR TITLE
feat(examples): Add simple example counter

### PR DESCRIPTION
**Pull request by briancavalier**
I figured I'd get the ball rolling on adding some examples.  This is the usual, trivial inc/dec counter.  It's probably more important to discuss:

1. Is this the right place to put examples?
2. Is the structure of the example reasonable? i.e. it's very self-contained, with it's own package.json, package-lock.json, and rollup.config.js.  Another option may be to share those one level up in the examples folder, as we've done in [mostjs/examples](https://github.com/mostjs/examples).  Might be easier maintenance if we do.

### CODE REVIEW

**Comment by davidchase** [comment]
What if we just offer the src/index.js and a link to a webpackbin where the viewer could play with the demo and also download the whole package that would be ready to work via a`yarn/npm install` that we its much less for us to maintain ?

File: examples/counter/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import node from 'rollup-plugin-node-resolve'
+import buble from 'rollup-plugin-buble'
+import flow from 'rollup-plugin-flow'
+
+export default {
+  input: 'src/index.js',
+  plugins: [
+    flow(),
+    buble(),
```
Comment (TrySound) [inline examples/counter/rollup.config.js]: Why buble for examples? es6 works great in the browsers and node.

File: examples/counter/src/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { click } from '@most/dom-event'
+
+const fail = s => { throw new Error(s) }
+const qs = (s) => document.querySelector(s) || fail(`${s} not found`)
+
+const incButton = qs('[name=inc]')
+const decButton = qs('[name=dec]')
+const value = qs('.value')
+
+const inc = map(_ => 1, click(incButton))
```
Comment (TrySound) [inline examples/counter/src/index.js]: I think this is the case for `constant` combinator.

File: examples/counter/rollup.config.js (hunk 1 above)
Comment (briancavalier) [reply examples/counter/rollup.config.js]: Sounds good to me.

**Comment by briancavalier** [comment]
@davidchase Can you say a bit more about exactly how your idea would work?  I'm generally in favor of anything that makes it easier for users to try the examples, as long as it doesn't add (much) maintenance or tooling complexity.

**Comment by davidchase** [comment]
After discussions I think we can for sake of not delaying v1 move with this example but ideally the setup would just require a JavaScript bin that supports flow or TS and we would just have a single EXAMPLES.md file that would link to these bins where developers could explore the examples as well as download the bin to run locally etc maybe something akin to https://hyperapp.glitch.me/ could work as well though I think glitch is still in beta and working out different bugs

**Pull request merged**